overlord/snapstate/snapstate.go

```diff
@@ -604,6 +604,9 @@ func checkInstallPreconditions(st *state.State, info *snap.Info, flags Flags, sn
 	if err := validateFeatureFlags(st, info); err != nil {
 		return err
 	}
+	if err := checkDBusServiceConflicts(st, info); err != nil {
```

Comment:
I fear there is a complication here, if you install many snaps "snap install a b" where both a and b activate on the same bus name we will not detect the conflict?

Reply:
You're right.  I've moved the check to the `link-snap` task and added a test to demonstrate that a concurrent install of conflicting snaps fails.